### DISCUSSION TITLE
HasOne nested key

### DISCUSSION THREAD

**Discussion by aqjw** [laravel/nova-issues#4487] [Q&A] [answered]
- Laravel Version: 9.11
- Nova Version: 4.10.0
- PHP Version: 8.1.3

### Description:

**HasOne** does not support nested keys.
As a result, it is not possible to read the value on the server-side.

<img width="1436" alt="Screenshot 2022-06-27 at 13 32 16" src="https://user-images.githubusercontent.com/13101908/175936480-0aaa24d8-9058-4917-b92e-8d9147e67d2b.png">

The keys are generated in the file `resources/js/fields/Form/HasOneField.vue`, but there is no nesting check here.

<img width="597" alt="Screenshot 2022-06-27 at 14 31 39" src="https://user-images.githubusercontent.com/13101908/175935972-fe263cc8-d254-43a5-84fd-787bafa84744.png">

### Solution 

<img width="766" alt="Screenshot 2022-06-27 at 14 32 20" src="https://user-images.githubusercontent.com/13101908/175936024-a49a8420-3b2a-46c4-b10e-a1cc709ca0cb.png">

```
entries.forEach((data, attribute) => {
  var attribute_key = `[${attribute}]`

  if (attribute.includes('[')) {
    let pos = attribute.indexOf('[')
    let key = attribute.substr(0, pos)
    let nested = attribute.substr(pos)
    attribute_key = `[${key}]${nested}` 
  }

  this.fillIfVisible(
    formData,
    this.field.attribute+attribute_key,
    data || ''
  )
})
```

**Comment by crynobone** [comment, marked as answer]
3rd party packages should instead submit custom attribute to `formData.append()` using `JSON.stringify()` to avoid such issue.
